Shell/Naming.cpp

```diff
@@ -317,14 +313,12 @@ Formula* Naming::apply_iter(Formula* top_f) {
               FormulaList::push(gs[i], rs);
             }
             f = new JunctionFormula(AND, rs);
-            DEALLOC_UNKNOWN(gs, "Naming::apply");
+            delete[] gs;
           } else if (fs != f->args()) {
             f = new JunctionFormula(AND, fs);
           }
-          DEALLOC_UNKNOWN(sand.cls, "Naming::apply");
-          if (sand.negCls) {
-            DEALLOC_UNKNOWN(sand.negCls, "Naming::apply");
-          }
+          delete[] sand.cls;
+          delete[] sand.negCls;
```

Comment:
I assume delete already checks for 0 / not-0, right?

Reply:
Right - standard apparently says that either (i) nothing will happen, or (ii) the deletion function will actually be called, but our deletion function (and that in the standard library) handle this case already. If not it's a bug there, not here.